crates/crypto/src/secp256k1.rs

```diff
@@ -1,20 +1,33 @@
 use {
-    crate::{to_sized, CryptoError, CryptoResult, Identity256},
+    crate::{to_sized, validate_length, CryptoError, CryptoResult, Identity256},
     k256::ecdsa::{signature::DigestVerifier, RecoveryId, Signature, VerifyingKey},
 };
 
-const SECP256K1_PUBKEY_LEN: usize = 33;
+const SECP256K1_PUBKEY_COMPRESSED_LEN: usize = 33;
+const SECP256K1_PUBKEY_UNCOMPRESSES_LEN: usize = 65;
 const SECP256K1_SIGNATURE_LEN: usize = 64;
 
 /// NOTE: This function takes the hash of the message, not the prehash.
 pub fn secp256k1_verify(msg_hash: &[u8], sig: &[u8], pk: &[u8]) -> CryptoResult<()> {
     let msg = Identity256::from_slice(msg_hash)?;
 
     let sig = to_sized::<SECP256K1_SIGNATURE_LEN>(sig)?;
-    let sig = Signature::from_bytes(&sig.into())?;
+    let mut sig = Signature::from_bytes(&sig.into())?;
+
+    // High-S signatures require normalization since our verification implementation
+    // rejects them by default. If we had a verifier that does not restrict to
+    // low-S only, this step was not needed.
+    if let Some(normalized) = sig.normalize_s() {
+        sig = normalized;
+    }
+
+    validate_length!(
+        pk,
+        SECP256K1_PUBKEY_COMPRESSED_LEN,
+        SECP256K1_PUBKEY_UNCOMPRESSES_LEN
+    )?;
```

Comment:
Yes, it's only to handle the error. It can be removed. Anyway the signature Error is always empty. It should be taken in consideration to properly emit a custom error depending on the failing reason